test/blackbox-tests/test-cases/pkg/ocamlformat/binary-avoid-conflict-with-project.t

```diff
@@ -0,0 +1,110 @@
+Make sure that the "dune-project" does not use the OCamlFormat binary dev-tool.
+
+  $ . ../helpers.sh
+  $ mkrepo
+
+Make a fake ocamlformat:
+  $ mkdir ocamlformat
+  $ cd ocamlformat
+  $ cat > dune-project <<EOF
+  > (lang dune 3.13)
+  > (package (name ocamlformat))
+  > EOF
+  $ cat > ocamlformat.ml <<EOF
+  > let version = "0.26.2"
+  > ;;
+  > let () = print_endline ("formatted with version "^version)
+  > EOF
+  $ cat > dune <<EOF
+  > (executable
+  >  (public_name ocamlformat))
+  > EOF
+  $ cd ..
+  $ tar -czf ocamlformat-0.26.2.tar.gz ocamlformat
+  $ rm -rf ocamlformat
+
+  $ mkpkg ocamlformat 0.26.2 <<EOF
+  > build: [
+  >   [
+  >     "dune"
+  >     "build"
+  >     "-p"
+  >     name
+  >     "@install"
+  >   ]
+  > ]
+  > url {
+  >  src: "file://$PWD/ocamlformat-0.26.2.tar.gz"
+  >  checksum: [
+  >   "md5=$(md5sum ocamlformat-0.26.2.tar.gz | cut -f1 -d' ')"
+  >  ]
+  > }
+  > EOF
+
+Make a project that uses the fake ocamlformat:
+  $ cat > dune-project <<EOF
+  > (lang dune 3.13)
+  > (package
+  >  (name foo))
+  > EOF
+  $ cat > foo.ml <<EOF
+  > let () = print_endline "Hello, world"
+  > EOF
+  $ cat > dune <<EOF
+  > (executable
+  >  (public_name foo))
+  > (rule
+  >  (target none)
+  >  (action
+  >     (progn
+  >       (run ocamlformat foo.ml)
+  >       (run touch none))))
+  > EOF
+  $ cat > dune-workspace <<EOF
+  > (lang dune 3.13)
+  > (lock_dir
+  >  (path "dev-tools.locks/ocamlformat")
+  >  (repositories mock))
+  > (lock_dir
+  >  (repositories mock))
+  > (repository
+  >  (name mock)
+  >  (source "file://$(pwd)/mock-opam-repository"))
+  > EOF
+
+Add a fake binary in the PATH
```

Comment:
```suggestion
Add a fake executable in the PATH
```
Let's reserve the word "binary" for actual binary files (ie. not for text files like scripts).